classes/submission/maps/Schema.php

```diff
@@ -526,16 +526,30 @@ protected function canChangeMetadata(Enumerable $stageAssignments, Submission $s
     /**
      * Get details about the review assignments for a submission
      */
-    protected function getPropertyReviewAssignments(Enumerable $reviewAssignments, bool|Collection $anonymizeReviews = false): array
+    protected function getPropertyReviewAssignments(Enumerable $reviewAssignments, bool|Collection $anonymizeReviews = false, Submission $submission): array
     {
-        $reviews = [];
         $request = Application::get()->getRequest();
         $currentUser = $request->getUser();
 
+        // current user's assigned roles from the submission's stages
+        $currentUserAssignedRoles = [];
+        foreach ($this->getPropertyStages($this->stageAssignments, $submission, $this->decisions ?? null, null) as $stage) {
+            if (isset($stage['currentUserAssignedRoles'])) {
```

Comment:
We are looking here for roles at the specific stage. For OJS it's `WORKFLOW_STAGE_ID_EXTERNAL_REVIEW`, for OMP it's `WORKFLOW_STAGE_ID_EXTERNAL_REVIEW` and `WORKFLOW_STAGE_ID_INTERNAL_REVIEW`